### PR TITLE
auth_ephemeral: fix authentication with secrets added at runtime

### PR DESCRIPTION
**Pull request by Den4t**
- proper way to work with shm pointer to secret list

#### Pre-Submission Checklist

- [x] Commit message has the format required by CONTRIBUTING guide
- [x] Commits are split per component (core, individual modules, libs, utils, ...)
- [x] Each component has a single commit (if not, squash them into one commit)
- [x] No commits to README files for modules (changes must be done to docbook files
in `doc/` subfolder, the README file is autogenerated)

#### Type Of Change
- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)

#### Checklist:

- [x] PR should be backported to stable branches
- [x] Tested changes locally
- [ ] Related to issue #XXXX (replace XXXX with an open issue number)

#### Description
In case of adding secrets at runtime with autheph.add_secret rpc, authentication with new secrets not work,
this is because rpc process modify local pointer to secret list, instead shm.

### CODE REVIEW

File: src/modules/auth_ephemeral/auth_ephemeral_mod.c [hunk 1]
```diff
@@ -226,12 +230,22 @@ static inline int add_secret(str _secret_key)
...
 	secret_struct->secret_key = _secret_key;
 	SECRET_LOCK;
-	if (secret_list != NULL)
+	if (secret_list == NULL)
+	{
+		secret_list = (struct secret **) shm_malloc(sizeof(struct secret *));
+		if (secret_list == NULL)
+		{
+			LM_ERR("unable to allocate shared memory\n");
+			return -1;
```
Comment (henningw) [inline src/modules/auth_ephemeral/auth_ephemeral_mod.c:239]: You probably want to free the previously allocated secret_struct

**Review by henningw — COMMENTED** [review]
Thanks for the PR. I just added one remark related to the freeing of the previous allocated memory in an error condition. Maybe others like to comment as well, as I do not use this module a lot.

File: src/modules/auth_ephemeral/auth_ephemeral_mod.c (hunk 1 above)
Comment (Den4t) [reply src/modules/auth_ephemeral/auth_ephemeral_mod.c:240]: Fixed.

**Comment by miconda** [comment]
Thanks!

**Pull request merged**